src/main/kotlin/com/openlattice/chronicle/services/surveys/SurveysService.kt

```diff
@@ -49,16 +93,91 @@ class SurveysService(
         TODO("Not yet implemented")
     }
 
-    override fun getParticipantAppsUsageData(
-            organizationId: UUID, studyId: UUID, participantId: String, date: String
-    ): List<ChronicleAppsUsageDetails> {
-        TODO("Not yet implemented")
-    }
 
-    override fun submitTimeUseDiarySurvey(
-            organizationId: UUID, studyId: UUID, participantId: String,
-            surveyData: List<Map<FullQualifiedName, Set<Any>>>
+    override fun submitAppUsageSurvey(
+            organizationId: UUID,
+            studyId: UUID,
+            participantId: String,
+            surveyResponses: Map<UUID, Set<String>>
     ) {
-        TODO("Not yet implemented")
+        logger.info(
+                "submitting app usage survey $ORG_STUDY_PARTICIPANT",
+                organizationId,
+                studyId,
+                participantId
+        )
+
+        val numWritten = updateAppUsage(organizationId, studyId, participantId, surveyResponses)
+
+        if (numWritten  != surveyResponses.size) {
+            logger.warn("updated {} entities but expected to update {} entities", numWritten, surveyResponses.size)
+        }
+    }
+
+    override fun getAppUsageData(organizationId: UUID, studyId: UUID, participantId: String, date: String): List<AppUsage> {
+        try {
+            val requestedDate = LocalDate.parse(date)
+
+            val (_, hds) = storageResolver.getDefaultStorage()
+
+            val result = BasePostgresIterable(
+                    PreparedStatementHolderSupplier(hds, GET_APP_USAGE_SQL) { ps ->
+                        ps.setObject(1, organizationId)
+                        ps.setObject(2, studyId)
+                        ps.setObject(3, participantId)
+                        ps.setObject(4, requestedDate)
+                    }
+            ) {
+                ResultSetAdapters.appUsage(it)
+
+            }.toList()
+
+            logger.info(
+                    "fetched {} app usage entries for date {} $ORG_STUDY_PARTICIPANT",
+                    result.size,
+                    requestedDate,
+                    organizationId,
+                    studyId,
+                    participantId
+            )
+
+            return result
+
+        } catch (ex: Exception) {
+            logger.error("unable to parse date string")
+            throw ex
+        }
+    }
+
+    private fun updateAppUsage( organizationId: UUID, studyId: UUID, participantId: String, data: Map<UUID, Set<String>>): Int {
+        val (_, hds) = storageResolver.resolveAndGetFlavor(studyId)
```

Comment:
Unless you are specifically planning to use redshift for survey storage, use the resolve with required flavor version of the call.